android/flavors/gms-enabled/com/mapswithme/maps/location/GoogleFusedLocationProvider.java

```diff
@@ -2,165 +2,126 @@
 
 import android.content.Context;
 import android.location.Location;
-import android.location.LocationListener;
-import android.os.Bundle;
+import android.os.Looper;
 
 import androidx.annotation.NonNull;
 
-import com.google.android.gms.common.ConnectionResult;
-import com.google.android.gms.common.api.GoogleApiClient;
-import com.google.android.gms.common.api.PendingResult;
-import com.google.android.gms.common.api.ResultCallback;
-import com.google.android.gms.common.api.Status;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationAvailability;
+import com.google.android.gms.location.LocationCallback;
 import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationResult;
 import com.google.android.gms.location.LocationServices;
 import com.google.android.gms.location.LocationSettingsRequest;
-import com.google.android.gms.location.LocationSettingsResult;
-import com.google.android.gms.location.LocationSettingsStatusCodes;
+import com.google.android.gms.location.SettingsClient;
 
 class GoogleFusedLocationProvider extends BaseLocationProvider
-    implements GoogleApiClient.ConnectionCallbacks,
-    GoogleApiClient.OnConnectionFailedListener
 {
   private final static String TAG = GoogleFusedLocationProvider.class.getSimpleName();
-  private final GoogleApiClient mGoogleApiClient;
-  private LocationRequest mLocationRequest;
-  private PendingResult<LocationSettingsResult> mLocationSettingsResult;
+  @NonNull
+  private final FusedLocationProviderClient mFusedLocationClient;
+  @NonNull
+  private final SettingsClient mSettingsClient;
 
-  private class GoogleLocationListener extends BaseLocationListener implements com.google.android.gms.location.LocationListener
+  private class GoogleLocationCallback extends LocationCallback
   {
-    private GoogleLocationListener(@NonNull LocationFixChecker locationFixChecker) {super(locationFixChecker);}
+    @Override
+    public void onLocationResult(@NonNull LocationResult result)
+    {
+      final Location location = result.getLastLocation();
```

Comment:
LocationResult.getLastLocation() is declared as NonNull.

```
  @NonNull
  public Location getLastLocation() {
    int var1 = this.zzb.size();
    return var1 == 0 ? null : (Location)this.zzb.get(var1 + -1);
  }
```

For `mFusedLocationClient.getLastLocation()` this check exists:

```
    mFusedLocationClient.getLastLocation().addOnSuccessListener(location -> {
      if (location == null)
        return;
```

Reply:
1. Documentation [says](https://developers.google.com/android/reference/com/google/android/gms/location/LocationResult?hl=en#getLastLocation()), that it may return null. It should crash on the very first cold start.
2. In your first code sample it returns null. Where did you get it?
3. How is the second code sample relevant?

Please fix.

Reply:
It looks like a bug in their SDK and/or documentation. I added an extra check for null. Thanks.